dbms/src/Encryption/FileProvider.cpp

```diff
@@ -196,6 +200,26 @@ void FileProvider::deleteEncryptionInfo(const EncryptionPath & encryption_path_,
     key_manager->deleteFile(encryption_path_.full_path, throw_on_error);
 }
 
+void FileProvider::encryptPage(
+    const EncryptionPath & encryption_path_,
+    char * data,
+    size_t data_size,
+    PageIdU64 page_id)
+{
+    const auto info = key_manager->getFile(encryption_path_.full_path);
```

Comment:
not related to this PR: Maybe rename `KeyManager::get/new/delete/linkFile` ->  `KeyManager::get/new/delete/linkInfo` ?

Reply:
lgtm, but leave it in next PR to keep #8585 more clean